elastalert/elastalert.py

```diff
@@ -190,10 +190,11 @@ def get_query(filters, starttime=None, endtime=None, sort=True, timestamp_field=
         starttime = to_ts_func(starttime)
         endtime = to_ts_func(endtime)
         filters = copy.copy(filters)
-        query = {'filter': {'bool': {'must': filters}}}
+        es_filters = {'filter': {'bool': {'must': filters}}}
         if starttime and endtime:
-            query['filter']['bool']['must'].append({'range': {timestamp_field: {'gt': starttime,
+            es_filters['filter']['bool']['must'].insert(0,{'range': {timestamp_field: {'gt': starttime,
```

Comment:
so that the timerange filter is applied first: https://www.elastic.co/guide/en/elasticsearch/guide/current/_filter_order.html

I guess it only makes a difference when the query is running against a huge index.